shap/benchmark/framework.py

```diff
@@ -0,0 +1,110 @@
+import matplotlib.pyplot as plt 
+import pandas as pd 
+import numpy as np 
+import sklearn 
+import shap 
+from sklearn.model_selection import train_test_split
+from shap.utils import safe_isinstance, MaskedModel
+from shap.benchmark import SequentialPerturbation
+
+def update(model, X, y, explainer, masker, sort_order, score_function, perturbation, scores):
+    metric = perturbation + ' ' + sort_order
+    sp = SequentialPerturbation(model, masker, sort_order, score_function, perturbation)
+    x, y, auc = sp.score(explainer, X, y=y)
+    scores['metrics'].append(metric)
+    scores['values'][metric] = [x, y, auc] 
+
+def get_benchmark(model, X, y, explainer, masker, metrics, exp_num=1, *args):
+    # convert dataframes
+    if safe_isinstance(X, "pandas.core.series.Series") or safe_isinstance(X, "pandas.core.frame.DataFrame"):
+        X = X.values
+    if safe_isinstance(masker, "pandas.core.series.Series") or safe_isinstance(masker, "pandas.core.frame.DataFrame"):
+        masker = masker.values
+    
+    # in case the explainer doesn't have a name
+    try: 
+        name = explainer.name 
+    except: 
+        name = 'explainer' + str(exp_num) 
+        exp_num += 1 
+
+    # record scores per metric 
+    scores = {'name': name, 'metrics': list(), 'values': dict()}
+    for sort_order in metrics['sort_order']:
+        for perturbation in metrics['perturbation']:
+            score_function = lambda true, pred: np.mean(pred)
+            update(model, X, y, explainer, masker, sort_order, score_function, perturbation, scores)
+
+    return scores 
+
+def get_metrics(benchmarks, selection):
+    # select metrics to plot using selection function
+    explainer_metrics = set()
+    for explainer in benchmarks: 
+        scores = benchmarks[explainer]
+        if len(explainer_metrics) == 0: explainer_metrics = set(scores['metrics'])
```

Comment:
put if/else body on separate line.
Hmm I think this is something flake8 would catch, are we running flake8 in the builds?

Reply:
I'm not sure if flake8 is being run but I've changed this.